apps/api/src/domain/models/playlist/commands/translate-playlist-name/translate-playlist-name.command-integration.spec.ts

```diff
@@ -0,0 +1,184 @@
+                    );
+
+                    expect(englishTextItem.text).toBe(englishName);
+                },
+            });
+        });
+    });
+
+    describe('when the command is invalid', () => {
+        describe('when the playlist does not exist', () => {
+            it('should fail with the expected errors', async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    buildCommandFSA: buildValidCommandFSA,
+                    initialState: new DeluxeInMemoryStore({}).fetchFullSnapshotInLegacyFormat(),
+                    checkError: (error) => {
+                        assertErrorAsExpected(
+                            error,
+                            new CommandExecutionError([
+                                new AggregateNotFoundError(
+                                    existingPlaylist.getCompositeIdentifier()
+                                ),
+                            ])
+                        );
+                    },
+                });
+            });
+        });
+
+        describe('when the original name is in the target translation language', () => {
+            it('should fail with the expected errors', async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    buildCommandFSA: buildValidCommandFSA,
+                    initialState: new DeluxeInMemoryStore({
+                        [AggregateType.playlist]: [
+                            existingPlaylist.clone({
+                                name: new MultilingualText({
+                                    items: [
+                                        new MultilingualTextItem({
+                                            text: 'I have the same language as the payload',
+                                            role: MultilingualTextItemRole.original,
+                                            // We invalidate the existing state to already have an original target translation language
+                                            languageCode: validPayload.languageCode,
+                                        }),
+                                    ],
+                                }),
+                            }),
+                        ],
+                    }).fetchFullSnapshotInLegacyFormat(),
+                    checkError: (error) => {
+                        assertErrorAsExpected(
+                            error,
+                            new CommandExecutionError([
+                                new DuplicateLanguageInMultilingualTextError(
+                                    validPayload.languageCode
+                                ),
+                            ])
+                        );
+                    },
+                });
```

Comment:
Don't forget to `Fuzz test` the command payload type.

Reply:
done